Bottom: 47c7589729c325b494b1f07f8eaa9e2daaa319aa
Top:    23ff296387a059a6b949a224aecfbdf288f93eb5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 19:29:45 +0000

Refresh of move-some-fields-from-the-hid

---

diff --git a/src/draw.c b/src/draw.c
index 45f72c0..f9fc2c7 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -176,10 +176,10 @@ _draw_pv_name (PinType *pv)
   text.Y = box.Y1;
   text.Direction = vert ? 1 : 0;
 
-  if (gui->graphics->klass->gui)
+  if (hid_draw_is_gui (gui->graphics))
     doing_pinout++;
   hid_draw_pcb_text (Output.fgGC, &text, 0);
-  if (gui->graphics->klass->gui)
+  if (hid_draw_is_gui (gui->graphics))
     doing_pinout--;
 }
 
@@ -320,7 +320,7 @@ pad_callback (const BoxType * b, void *cl)
 static void
 draw_element_name (ElementType *element)
 {
-  if ((TEST_FLAG (HIDENAMESFLAG, PCB) && gui->graphics->klass->gui) ||
+  if ((TEST_FLAG (HIDENAMESFLAG, PCB) && hid_draw_is_gui (gui->graphics)) ||
       TEST_FLAG (HIDENAMEFLAG, element))
     return;
   if (doing_pinout || doing_assy)
@@ -591,11 +591,11 @@ DrawEverything (const BoxType *drawn_area)
         }
     }
 
-  if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->graphics->klass->gui)
+  if (TEST_FLAG (CHECKPLANESFLAG, PCB) && hid_draw_is_gui (gui->graphics))
     return;
 
   /* Draw pins, pads, vias below silk */
-  if (gui->graphics->klass->gui)
+  if (hid_draw_is_gui (gui->graphics))
     DrawPPV (SWAP_IDENT ? bottom_group : top_group, drawn_area);
   else
     {
@@ -639,7 +639,7 @@ DrawEverything (const BoxType *drawn_area)
       hid_draw_end_layer (gui->graphics);
     }
 
-  if (gui->graphics->klass->gui)
+  if (hid_draw_is_gui (gui->graphics))
     {
       /* Draw element Marks */
       if (PCB->PinOn)
@@ -740,7 +740,7 @@ DrawPPV (int group, const BoxType *drawn_area)
   int bottom_group = GetLayerGroupNumberBySide (BOTTOM_SIDE);
   int side;
 
-  if (PCB->PinOn || !gui->graphics->klass->gui)
+  if (PCB->PinOn || !hid_draw_is_gui (gui->graphics))
     {
       /* draw element pins */
       r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_callback, NULL);
@@ -760,7 +760,7 @@ DrawPPV (int group, const BoxType *drawn_area)
     }
 
   /* draw vias */
-  if (PCB->ViaOn || !gui->graphics->klass->gui)
+  if (PCB->ViaOn || !hid_draw_is_gui (gui->graphics))
     {
       r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
       r_search (PCB->Data->via_tree, drawn_area, NULL, hole_callback, NULL);
@@ -974,7 +974,7 @@ DrawLayer (LayerType *Layer, const BoxType *screen)
   /* draw the layer text on screen */
   r_search (Layer->text_tree, screen, NULL, text_callback, Layer);
 
-  /* We should check for gui->graphics->klass->gui here, but it's kinda cool seeing the
+  /* We should check for hid_draw_is_gui (gui->graphics) here, but it's kinda cool seeing the
      auto-outline magically disappear when you first add something to
      the "outline" layer.  */
   if (IsLayerEmpty (Layer)
@@ -1013,7 +1013,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
   if (n_entries > 1)
     rv = 1;
 
-  if (rv && !gui->graphics->klass->gui)
+  if (rv && !hid_draw_is_gui (gui->graphics))
     DrawPPV (group, drawn_area);
 }
 
diff --git a/src/hid_draw.h b/src/hid_draw.h
index 64cb7ec..a442390 100644
--- a/src/hid_draw.h
+++ b/src/hid_draw.h
@@ -100,6 +100,15 @@ struct hid_gc_struct {
   HID_DRAW *hid_draw;
 };
 
+
+/* Getters for class properties */
+inline bool
+hid_draw_is_gui (HID_DRAW *hid_draw)
+{
+  return hid_draw->klass->gui;
+}
+
+
 /* Calling wrappers to access the vfunc table */
 
 inline int
